ts/Series/GeoHeatmap/GeoHeatmapSeries.ts

```diff
@@ -175,9 +381,314 @@ class GeoHeatmapSeries extends MapSeries {
+                ],
+                // Calculate dimensions based on series bounds
+                topLeft = mapView.projectedUnitsToPixels({
+                    x: series.bounds.x1,
+                    y: series.bounds.y2
+                }),
+                bottomRight = mapView.projectedUnitsToPixels({
+                    x: series.bounds.x2,
+                    y: series.bounds.y1
+                });
+
+            if (canvas && ctx && colorAxis && topLeft && bottomRight) {
+                const dimensions = {
+                    x: topLeft.x,
+                    y: topLeft.y,
+                    width: bottomRight.x - topLeft.x,
+                    height: bottomRight.y - topLeft.y
+                };
+
+                if (
+                    // Do not calculate new canvas if not necessary
+                    series.isDirtyCanvas ||
+                    // Calculate new canvas if data is dirty
+                    series.isDirtyData ||
+                    // Always calculate new canvas for Orthographic projection
+                    mapView.projection.options.name === 'Orthographic'
+                ) {
+                    series.isDirtyCanvas = true;
+                    const canvasWidth = canvas.width = ~~(360 / colsize) + 1,
+                        canvasHeight = canvas.height = ~~(180 / rowsize) + 1,
+                        canvasArea = canvasWidth * canvasHeight,
+                        pixelData = new Uint8ClampedArray(canvasArea * 4);
+                    series.directTouch = false; // Needed for tooltip
+
+                    // First pixelData represents the geo coordinates
+                    for (let i = 0; i < points.length; i++) {
+                        const p = points[i],
+                            sourceArr = new Uint8ClampedArray(
+                                colorFromPoint(p.value, p)
+                            ),
+                            { lon, lat } = p.options;
+
+                        if (isNumber(lon) && isNumber(lat)) {
+                            pixelData.set(
+                                sourceArr,
+                                scaledPointPos(
+                                    lon, lat, canvasWidth, canvasHeight,
+                                    colsize, rowsize
+                                ) * 4
+                            );
+                        }
+                    }
+
+                    const blur = pick(series.options.interpolationBlur, 1),
+                        blurFactor = blur === 0 ? 1 : blur * 11;
+
+
+                    const upscaledWidth = ~~(canvasWidth * blurFactor),
+                        upscaledHeight = ~~(canvasHeight * blurFactor),
+                        projectedWidth = ~~dimensions.width,
+                        projectedHeight = ~~dimensions.height;
+
+                    canvas.width = upscaledWidth;
+                    canvas.height = upscaledHeight;
+
+                    const img =
```

Comment:
Looks like these lines could be cleaned-up - e.g. less const, less empty lines, etc.